### PR TITLE
Code optimisation of admin side done.

### PR DESCRIPTION
**Pull request by Pramodjais517**
Solving #3 
1. Admin Side optimization including CRUD operation of : 

- [x]  Test 
- [x] Category
- [x]  Question
- [x]  Instructions.

### CODE REVIEW

**Comment by Pramodjais517** [comment]
@deepanshu-jain1999 @bvishal8510 closed the previous PR due to branch merge issue.
This PR contains the required changes.

**Review by SukhnandanMalhotra — APPROVED** [review]

File: quiz_portal/settings/base.py [hunk 1]
```diff
@@ -135,6 +135,7 @@
 CKEDITOR_RESTRICT_BY_USER = True
 CKEDITOR_ALLOW_NONIMAGE_FILES = False
 
-RECAPTCHA_PRIVATE_KEY = os.environ['RECAPTCHA_PRIVATE_KEY']
-RECAPTCHA_PUBLIC_KEY = os.environ['RECAPTCHA_PUBLIC_KEY']
+RECAPTCHA_PRIVATE_KEY = os.environ.get('GOOGLE_RECAPTCHA_SECRET_KEY')
```
Comment (bvishal8510) [inline quiz_portal/settings/base.py]: why changed the name?

File: quiz_portal/settings/base.py [hunk 2]
```diff
@@ -135,6 +135,7 @@
...
+RECAPTCHA_PUBLIC_KEY = os.environ.get('GOOGLE_RECAPTCHA_PUBLIC_KEY')
```
Comment (bvishal8510) [inline quiz_portal/settings/base.py]: here too, revert it as it was before

File: core/models.py [hunk 3]
```diff
@@ -58,26 +57,26 @@ class Question(models.Model):
...
-    std_no = models.CharField(blank=True, max_length=7, null=True)
-    university_roll_no = models.CharField(max_length=10, unique=True)
+    std_no_regex = RegexValidator(regex=r"^\d{7}$", message="Invalid Student Number", code="400")
+    std_no = models.CharField(validators=[std_no_regex], blank=True, max_length=7, null=True)
+    university_roll__no_regex = RegexValidator(regex=r'^[1]\d{9}$',message="invalid university roll no.")
+    university_roll_no = models.CharField(max_length=10,blank=True, unique=True,validators=[university_roll__no_regex],null=True)
     email = models.EmailField(unique=True, blank=False)
     father = models.CharField(max_length=255, blank=True, null=True)
-    phone_regex = RegexValidator(regex=r"^[789]\d{9}$")
+    phone_regex = RegexValidator(regex=r"^[6789]\d{9}$")
```
Comment (bvishal8510) [inline core/models.py]: This regex won't accept all kinds of valid phone numbers. Now numbers start from 5 also. Use a valid regex.

Comment (Pramodjais517) [reply core/models.py]: I improved the previous ones and added 6. Ok I will add 5 also

Comment (bvishal8510) [reply core/models.py]: search for a better regex instead of just adding 5.

File: quiz_portal/settings/base.py (hunk 1 above)
Comment (Pramodjais517) [reply quiz_portal/settings/base.py]: Sir, I saved my previous ReCaptcha using this name only. No problem I will change it.

File: core/models.py (hunk 3 above)
Comment (Stark-Pramod) [reply core/models.py]: > This regex won't accept all kinds of valid phone numbers. Now numbers start from 5 also. Use a valid regex.

Sir no number starts from 5 at least not this year. So I think for security purpose this is ok.

File: quiz_portal/settings/base.py (hunk 1 above)
Comment (Pramodjais517) [reply quiz_portal/settings/base.py]: Done

**Comment by Pramodjais517** [comment]
@bvishal8510 Changes Done.

**Pull request merged**